www/js/onboarding/WelcomePage.tsx

```diff
@@ -0,0 +1,202 @@
+import React, { useContext, useState } from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+import { View, Image, Modal, ScrollView, StyleSheet, ViewStyle, useWindowDimensions } from 'react-native';
+import { Button, Dialog, Divider, IconButton, Surface, Text, TextInput, TouchableRipple, useTheme } from 'react-native-paper';
+import color from 'color';
+import { initByUser } from '../config/dynamicConfig';
+import { AppContext } from '../App';
+import { displayError } from "../plugin/logger";
+import { onboardingStyles } from './OnboardingStack';
+import { Icon } from '../components/Icon';
+
+const WelcomePage = () => {
+
+  const { t } = useTranslation();
+  const { colors } = useTheme();
+  const { width: windowWidth } = useWindowDimensions();
+  const context = useContext(AppContext);
+  const { refreshOnboardingState } = context;
+  const [pasteModalVis, setPasteModalVis] = useState(false);
+  const [infoPopupVis, setInfoPopupVis] = useState(false);
+  const [existingToken, setExistingToken] = useState('');
+
+  const scanCode = function() {
+    window.cordova.plugins.barcodeScanner.scan(
+      function (result) {
+        console.debug("scanned code", result);
+          if (result.format == "QR_CODE" && 
+              result.cancelled == false) {
+                let text = result.text.split("=")[1];
+                console.log("found code", text);
+              loginWithToken(text);
```

Comment:
weren't we checking here to see that the QR code actually did start with `emission://`?

Reply:
That was somewhere in the older version of the QR scanning flow, you're right. I think I got focused on splitting the token from the rest of the url, I can add that back in the cleanup pass. Before, there was a line with `result.text.startsWith("emission://"))`